.dev/script/new-traffic-data_20210508.R

```diff
@@ -0,0 +1,228 @@
+  )
+
+# data clean --------------------------------------------------------------
+# this chunk does two things:
+# 1. join new variables up to the old datasets
+# 2. "code" new variables as factor / categorical variables
+
+look_up <- function(x,
+                    dictionary,
+                    index = "Code",
+                    match = "Description"){
+  dictionary[[match]][c(match(x, dictionary[[index]]))]
+}
+
+# look_up(iris$Species,
+#         c("virginica", "versicolor", "setosa"),
+#         replace = c("tum", "de", "da"))
+
+hk_accidents_new_cleaned_unlabelled <-
+  hk_accidents %>%
+  left_join(
+    # New columns
+    select(
+      hk_accidents_new,
+      
+      # variables for joining ---------------------------------------
+      Date, # For joining
+      Serial_No_, # For joining
+      Year, # For joining
+      
+      # New variables with code frames identified --------------------
+      Street_nam, # Street name, need to be joined with A1_street_name
+      Second_str, # Second street name, need to be joined with A1_street_name
+      Overtaking, # E.g. One vehicle, two or more, etc.
+      Within_70m, # Whether within 70m of junction
+      Road_type_, # E.g. One way, Two way, Dual carriageway
+      Pedal_cycl, # Others, Pedal Cycle, Motorcycle
+      TypeOfCo_P, # E.g. Vehicle collision with Pedestrian, etc.
+      Struc_Type, # E.g. At grade road, Flyover, Bridge, etc.
+      RD_Class_L, # E.g. Expressway, Main Road, Secondary Road, etc.
+      Road_class, # Slightly different from `Road_Classification`
+      
+      # New variables that do not require code frame -----------------
+      Precise_lo, # Precise location. Strength.
+      Accident_a, # No/ Yes
+      Junction_t, # E.g. Cross roads, T-junction, Y-junction, Multiple, etc.
+      Whether_at, # E.g. No crossing control, On a crossing control, etc.
+      Type_of_Cr, # E.g. Footbridge/Subway, Traffic signal, Zebra, etc.
+      
+      ),
+    by = c("Date", "Serial_No_", "Year")
+  )
+  
+  # labelling ----------------------------------------------------------------
+
+hk_accidents_new_cleaned_labelled <-
+  hk_accidents_new_cleaned_unlabelled %>%
+  mutate(Street_Name = look_up(Street_nam, dictionary = t_A1_street_name)) %>%
+  mutate(Second_Street_Name = look_up(Second_str, dictionary = t_A1_street_name)) %>%
+  mutate(Overtaking = look_up(Overtaking, dictionary = t_Overtaking)) %>%
+  mutate(Within_70m = look_up(Within_70m, dictionary =  t_Within_70m)) %>%
+  mutate(Road_Type = look_up(Road_type_, dictionary = t_Road_type_)) %>%
+  mutate(Cycle_Type = look_up(Pedal_cycl, dictionary = t_Ped_Cycle)) %>%
+  mutate(Type_of_Collision_v2 = look_up(TypeOfCo_P, dictionary = t_Collision_Type)) %>%
+  mutate(Structure_Type = look_up(Struc_Type, dictionary = t_Struc_Type)) %>%
+  mutate(Road_Class_L = look_up(RD_Class_L, dictionary = t_Road_class_L)) %>%
+  mutate(Road_Classification_v2 = look_up(Road_class, dictionary = t_Road_class)) %>%
```

Comment:
Could I know the reasons for separating the `mutate` function one by one? I usually put all new columns to be created into  one single `mutate` function like

```r
mutate(
  Street_Name = look_up(Street_nam, dictionary = t_A1_street_name),
  Second_Street_Name = look_up(Second_str, dictionary = t_A1_street_name),
  Overtaking = look_up(Overtaking, dictionary = t_Overtaking),
  ...
  Road_Classification_v2 = look_up(Road_class, dictionary = t_Road_class)
)
```

Reply:
Good shout - @KHwong12 your suggestion is actually best practice, so I'll commit that now.